### PR TITLE
Modify changelog generator functionality

### PR DESCRIPTION
**Pull request by jasiu001**
Upgrading changelog generator tool so that it can be used in others repositories.

### CODE REVIEW

File: tools/changelog-generator/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,5 @@
 FROM node:10.11.0-alpine
 
-LABEL source git@github.com:kyma-project/kyma.git
```
Comment (pkosiec) [inline tools/changelog-generator/Dockerfile]: The label shouldn't be deleted

File: release.Jenkinsfile [hunk 2]
```diff
@@ -163,31 +163,55 @@ try {
...
+                                [string(credentialsId: 'public-github-token', variable: 'token'),
+                                sshUserPrivateKey(credentialsId: "bitbucket-rw", keyFileVariable: 'sshfile')
+                        ]) {
+                            // Build changelog generator
+                            dir(changelogGeneratorPath) {
+                                sh "docker build -t changelog-generator ."
+                            }
+
+                            // Setup path to repository config file for `lerna-changelog`
+                            configFile = "./tools/changelog-generator/package.json"
```
Comment (pkosiec) [inline release.Jenkinsfile]: Use `def`
https://stackoverflow.com/questions/184002/groovy-whats-the-purpose-of-def-in-def-x-0

File: tools/changelog-generator/README.md [hunk 3]
```diff
@@ -27,14 +27,17 @@ To generate a `CHANGELOG.md` file for all released Kyma versions, follow these s
...
 ```
 
 Replace values in curly braces with proper details, where:
 - `{absolutePathToRepository}` is the absolute path to the repository.
 - `{applicationVersion}` is the currently released application version.
 - `{githubToken}` is the GitHub API token with the read-only access to the repository.
 - `{sshFilePath}` is the path to the SSH file used for Git to authenticate with the repository.
+- `{configFilePath}` is the path to file used by `lerna-changelog` to repository configuration
+
+Add `SKIP_REMOVING_LATEST=true` variable if you want skip removing 'latest' tag functionality.
```
Comment (pkosiec) [inline tools/changelog-generator/README.md]: Please describe configuration for `SKIP_REMOVING_LATEST like in above paragraph.

So in the example above it should be something like 
`-e SKIP_REMOVING_LATEST={skipRemovingLatest}` and then below you describe the role of the variable

File: tools/changelog-generator/README.md [hunk 4]
```diff
@@ -53,14 +56,18 @@ Replace values in curly braces with proper details, where:
...
 
 Replace values in curly braces with proper details, where:
 - `{absolutePathToRepository}` is the absolute path to the repository.
+- `{previousTag}` optionally, one tag before last. If provided changelog will be generated from `{previousTag}` to `{applicationVersion}`
 - `{applicationVersion}` is the currently released application version.
 - `{githubToken}` is the GitHub API token with the read-only access to the repository.
 - `{sshFilePath}` is the path to the SSH file used for Git to authenticate with the repository.
+- `{configFilePath}` is the path to file used by `lerna-changelog` to repository configuration
+
+Add `SKIP_REMOVING_LATEST=true` variable if you want skip removing 'latest' tag functionality.
```
Comment (pkosiec) [inline tools/changelog-generator/README.md]: Please describe configuration for `SKIP_REMOVING_LATEST like in above paragraph.

So in the example above it should be something like 
`-e SKIP_REMOVING_LATEST={skipRemovingLatest}` and then below you describe the role of the variable

File: tools/changelog-generator/README.md [hunk 5]
```diff
@@ -53,14 +56,18 @@ Replace values in curly braces with proper details, where:
 To generate a changelog for a single release that contains merged pull requests for the latest Kyma version, run this command:
 
 ```bash
-docker run --rm -v /path/to/repository/:/repository -w /repository -e LATEST_VERSION={applicationVersion} -e GITHUB_AUTH={githubToken} -e SSH_FILE={sshFile} changelog-generator sh /app/generate-release-changelog.sh --configure-git
+docker run --rm -v /path/to/repository/:/repository -w /repository -e FROM_TAG={previousTag} -e LATEST_VERSION={applicationVersion} -e GITHUB_AUTH={githubToken} -e SSH_FILE={sshFile} -e CONFIG_FILE={configFilePath} changelog-generator sh /app/generate-release-changelog.sh --configure-git
 ```
 
 Replace values in curly braces with proper details, where:
 - `{absolutePathToRepository}` is the absolute path to the repository.
+- `{previousTag}` optionally, one tag before last. If provided changelog will be generated from `{previousTag}` to `{applicationVersion}`
```
Comment (pkosiec) [inline tools/changelog-generator/README.md]: It should be mentioned, that if not provided, the latest tag is used.

File: tools/changelog-generator/README.md [hunk 6]
```diff
@@ -27,14 +27,17 @@ To generate a `CHANGELOG.md` file for all released Kyma versions, follow these s
 1. Create a `CHANGELOG.md` file under the specified absolute path to the repository. Use this command:
 
 ```bash
-docker run --rm -v {absolutePathToRepository}:/repository -w /repository -e LATEST_VERSION={applicationVersion} -e GITHUB_AUTH={githubToken} -e SSH_FILE={sshFilePath} changelog-generator sh /app/generate-full-changelog.sh --configure-git
+docker run --rm -v {absolutePathToRepository}:/repository -w /repository -e LATEST_VERSION={applicationVersion} -e GITHUB_AUTH={githubToken} -e SSH_FILE={sshFilePath} -e CONFIG_FILE={configFilePath} -e SKIP_REMOVING_LATEST=true changelog-generator sh /app/generate-full-changelog.sh --configure-git
```
Comment (pkosiec) [inline tools/changelog-generator/README.md]: Please rename `LATEST_VERSION` to something like `NEW_RELEASE_TITLE` or something like that - as it is only used for title in changelog, right?

File: tools/changelog-generator/app/config-setup.sh [hunk 7]
```diff
@@ -1,14 +1,19 @@
...
 
-LATEST_TAG_EXISTS=false
-if [ $(git tag -l "$LATEST_TAG") ]; then
-    LATEST_TAG_EXISTS=true
-    echo "Temporary removing 'latest' tag..."
-    LATEST_TAG_REV=$(git rev-list -n 1 ${LATEST_TAG})
-    LATEST_TAG_MESSAGE=$(git tag -l --format='%(contents)' ${LATEST_TAG})
-    git tag -d ${LATEST_TAG}
-fi
+function removingLatestTag() {
```
Comment (pkosiec) [inline tools/changelog-generator/app/config-setup.sh]: `removeLatestTag` would be a better name for a function I think

File: tools/changelog-generator/app/generate-release-changelog.sh [hunk 8]
```diff
@@ -9,10 +9,15 @@ fi
...
-REVISION=$(git rev-list --tags --max-count=1)
-LAST_VERSION_TAG=$(git describe --tags ${REVISION})
-echo "Getting new changes starting from the '${LAST_VERSION_TAG}' tag..."
 mkdir -p ${RELEASE_CHANGELOG_FILE_DIRECTORY}
-lerna-changelog --from=${LAST_VERSION_TAG} | sed -e "s/## Unreleased/## ${LATEST_VERSION}/g" > ${RELEASE_CHANGELOG_FILE_PATH}
+if [ -n "$FROM_TAG" ]; then
+    echo "Getting new changes starting from the '${FROM_TAG}' tag to '${LATEST_VERSION}' tag..."
+    lerna-changelog --from=${FROM_TAG} --to=${LATEST_VERSION} | sed -e "s/## Unreleased/## ${LATEST_VERSION}/g" > ${RELEASE_CHANGELOG_FILE_PATH}
+else
+    REVISION=$(git rev-list --tags --max-count=1)
```
Comment (pkosiec) [inline tools/changelog-generator/app/generate-release-changelog.sh]: You can simplify this fragment greatly. Here's the pseudocode:

```
if ($FROM_TAG) {
TAG = FROM_TAG
} else {
REVISION=$(git rev-list --tags --max-count=1)
LAST_VERSION_TAG=$(git describe --tags ${REVISION});
TAG=LATEST_VERSION_TAG
}

echo "Getting new changes starting from the '${TAG}' tag..."
    lerna-changelog --from=${TAG} | sed -e "s/## Unreleased/## ${LATEST_VERSION}/g" > ${RELEASE_CHANGELOG_FILE_PATH}

```

File: tools/changelog-generator/README.md (hunk 6 above)
Comment (jasiu001) [reply tools/changelog-generator/README.md]: It is used to changelog file tittle, in commit message (during pull changelog file to root repository) and if "FROM_TAG" is used then learna-changelog got it as a "to" argument. I changed the name.

File: tools/changelog-generator/app/generate-release-changelog.sh (hunk 8 above)
Comment (jasiu001) [reply tools/changelog-generator/app/generate-release-changelog.sh]: Another lines are also different. In first condition lerna-changelog is run with "from" and "to" arguments, in second condition only with "from". Echo messages are also different.

File: tools/changelog-generator/README.md (hunk 6 above)
Comment (pkosiec) [reply tools/changelog-generator/README.md]: > if "FROM_TAG" is used then learna-changelog got it as a "to" argument

Hmm, but to be honest it should be used in this way, as by design the tag shouldn't exist yet 😛

File: tools/changelog-generator/README.md [hunk 9]
```diff
@@ -27,40 +27,46 @@ To generate a `CHANGELOG.md` file for all released Kyma versions, follow these s
...
 
 ```bash
-docker run --rm -v /path/to/repository/:/repository -w /repository -e LATEST_VERSION={applicationVersion} -e GITHUB_AUTH={githubToken} -e SSH_FILE={sshFile} changelog-generator sh /app/generate-release-changelog.sh --configure-git
+docker run --rm -v {absolutePathToRepository}:/repository -w /repository -e FROM_TAG={previousTag} -e TO_TAG={latestTag} -e NEW_RELEASE_TITLE={releaseTitle} -e GITHUB_AUTH={githubToken} -e SSH_FILE={sshFile} -e CONFIG_FILE={configFilePath} -e SKIP_REMOVING_LATEST={skipRemovingLatest} changelog-generator sh /app/generate-release-changelog.sh --configure-git
 ```
 
 Replace values in curly braces with proper details, where:
 - `{absolutePathToRepository}` is the absolute path to the repository.
-- `{applicationVersion}` is the currently released application version.
+- `{previousTag}` optionally, one tag before last. If provided changelog will be generated from `{previousTag}` to `{latestTag}`. If not provided changelog will be generated from the one tag before latest used tag.
```
Comment (klaudiagrz) [inline tools/changelog-generator/README.md]:
```suggestion
- `{previousTag}` specifies the first tag which is included in the changelog. If not provided, the changelog is generated from the tag which precedes the latest tag used.
```

File: tools/changelog-generator/README.md [hunk 10]
```diff
@@ -27,40 +27,46 @@ To generate a `CHANGELOG.md` file for all released Kyma versions, follow these s
...
+- `{latestTag}` optionally, is tag to which changelog is created. If provided changelog will be generated to `{latestTag}`. If not provided changelog will be generated to latest used tag.
```
Comment (klaudiagrz) [inline tools/changelog-generator/README.md]:
```suggestion
- `{latestTag}` specifies the last tag which is included in the changelog. If not provided, the changelog is generated to the latest tag used.
```

**Pull request merged**